web/src/ApiButton.tsx

```diff
@@ -307,6 +334,41 @@ async function updateButtonStatus(
   await tiltApiPut("uibuttons", "status", toUpdate)
 }
 
+function getButtonTags(button: UIButton): Tags {
+  // The location of the button in the UI, either `Nav` or `Resource`
+  const component = button.spec?.location?.componentType
+
+  const buttonAnnotations = annotations(button)
+
+  // A unique hash of the button text to help identify which button was clicked
+  const specHash = buttonAnnotations[UIBUTTON_SPEC_HASH]
+
+  // Tilt-specific button annotation, currently only used to differentiate disable toggles
+  const buttonType = buttonAnnotations[UIBUTTON_ANNOTATION_TYPE]
+
+  // A toggle button will have a hidden input field with the current value of the toggle
+  // e.g., when a disable button is clicked, the hidden input will be "on" because that's
+  // the toggle's value when it's clicked, _not_ the value it's being toggled to.
+  let toggleInput: UIInputSpec | undefined
+  if (button.spec?.inputs) {
+    toggleInput = button.spec.inputs.find(
+      (input) => input.name === UIBUTTON_TOGGLE_INPUT_NAME
+    )
+  }
+
+  let toggleValue: ApiButtonToggleState | undefined
+  if (toggleInput !== undefined) {
+    toggleValue = toggleInput.hidden?.value as ApiButtonToggleState
```

Comment:
per documentation here:
https://docs.tilt.dev/telemetry_faq.html#what-kinds-of-data-does-tilt-record-if-i-opt-in

i think we want to avoid sending personalized user data in anonymized tags. instead, can you whitelist the values that you want to send? like "on" / "off"?

Reply:
ah, yes! i didn't realize there could be other values there.